main_server.py

```
@@ -647,80 +692,131 @@ async def p2p_info():
+                        status_code=503
+                    )
+
+        except (httpx.ConnectError, httpx.TimeoutException) as e:
+            last_error = e
+            logger.debug(f"LAN proxy connection attempt {attempt + 1}/{max_retries} failed: {e}")
+            if attempt < max_retries - 1:
+                import asyncio
+                await asyncio.sleep(0.5)  # 等待后重试
+            continue
+
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)